### PR TITLE
practical 1 and 2 work 

### PR DESCRIPTION
**Pull request by abbyhid00**

### CODE REVIEW

File: work/Abigail/excercise_1.1.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import os
+from os import fork
+
+pid = fork()
+
+if pid < 0:
+    print("Error in fork")
+elif pid == 0: #child process
+    time.sleep(2)
+    print("Hello")
```
Comment (Alvaro8gb) [inline work/Abigail/excercise_1.1.py]: Its good but is better do exec: 
tip: command = ["python", "-c", "import time; time.sleep(2); print('Hello')"]

File: work/Abigail/excercise_1.2.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    command = sys.argv[1]
+    
+    # Fork the process
+    pid = os.fork()
+    if pid < 0:
+        print("Fork failed")
+        sys.exit(1)
+    elif pid == 0:  # Child process
+        # Execute the command using exec
+        subprocess.run(command, shell=True)
```
Comment (Alvaro8gb) [inline work/Abigail/excercise_1.2.py]: this fuction of subprocess do inside a fork and a exec : its better only do a exec

File: work/Abigail/Practical 1.3.png
Comment (Alvaro8gb) [inline work/Abigail/Practical 1.3.png:1]: to see in unix machine if the is a zombine you can execute the program in the background with & and the in the cmd do :  ps aux | grep 'Z'

File: work/Abigail/practical_work_2.md [hunk 3]
```diff
@@ -10,14 +10,18 @@ The first child process forks another child process and them decrements the valu
 
 c) Draw the process tree resulting from the execution of run_processes 3.
 
-
-COME BACK TO THIS QUESTION
-
+Parent
+|
+|
```
Comment (Alvaro8gb) [inline work/Abigail/practical_work_2.md:15]: will check this in class

File: work/Abigail/practical_work_2.md [hunk 4]
```diff
@@ -10,14 +10,18 @@ The first child process forks another child process and them decrements the valu
...
+The total number of processes created during the execution of run_processes 3 would be 2.
 
 e) Reason whether, by executing run_processes 3, there could be any orphaned processes and/or any zombie processes. An orphaned process is considered to be one whose parent has died and is adopted by the init process.
 
-In this script there could have been an orphaned process created if line 25 in the code was not present. This forces the parent process to wait for the child process, so that the child does not get orphaned. Additionally, if line 25 was not present a zombie process could be created because the parent process would fail to wait for the child. However, with the script the way it is now there could not be any orphaned or zombie processes.
+In this code there could have been an orphaned process created if line 25 in the code was not present. This forces the parent process to wait for the child process, so that the child does not get orphaned. Additionally, if line 25 was not present a zombie process could be created because the parent process would fail to wait for the child. However, with the script the way it is now there could not be any orphaned or zombie processes.
```
Comment (Alvaro8gb) [inline work/Abigail/practical_work_2.md:26]: good

File: work/Abigail/excercise_1.2.py (hunk 2 above)
Comment (abbyhid00) [reply work/Abigail/excercise_1.2.py]: In the most recent commit "final additions to code and updating to use correct command" I changed this to use exec instead

**Pull request closed without merging**